src/CommunicationExecutor.c

```diff
@@ -61,11 +61,29 @@ void Ros_Communication_ConnectToAgent()
 
     //Construct the RMW Micro-XRCE-DDS client key to use
     //Use MAC last 4 bytes as client key (1 byte from the OUI, 3 bytes from NIC)
+    //TODO(gavanderhoorn): get MAC only from interface which is used for
+    //ROS traffic (https://github.com/Yaskawa-Global/motoros2/issues/57)
     UCHAR macId[6];
-    motoRosAssert_withMsg(
-        Ros_GetMacAddress(macId) == OK,
-        SUBCODE_FAIL_MP_NICDATA,
-        "Must enable ETHERNET function");
+    STATUS status = Ros_GetMacAddress(ROS_USER_LAN1, macId);
+    if (status != OK)
+    {
+        Ros_Debug_BroadcastMsg("%s: Ros_GetMacAddress: iface: %d; error: %d",
+            __func__, ROS_USER_LAN1, status);
+        motoRosAssert_withMsg(FALSE, SUBCODE_FAIL_MP_NICDATA_INIT0,
```

Comment:
On YRC, this won't allow the code to get to the point where it checks LAN2

Reply:
I know. But do we want to just try any interface which works, or should we actually just raise an alarm if we can't retrieve the MAC for the interface we're asking for?

I would probably suggest to do the first, as the previous behaviour could lead to situations where MotoROS2 ends up trying to use the 3rd interface while the 2nd is actually the one with the network cable (ie: for some reason, 2nd interface MAC can't be read, 3rd interface succeeds -> we just continue with that interface).

Reply:
> I would probably suggest to do the first, as the previous behaviour could lead to situations where MotoROS2 ends up trying to use the 3rd interface while the 2nd is actually the one with the network cable (ie: for some reason, 2nd interface MAC can't be read, 3rd interface succeeds -> we just continue with that interface).

I don't understand this comment. I'm not sure which of the two you're suggesting.

Personally, I'd just go with any interface that works.

Also, if we do maintain this assertion, then the check for the second MAC will never logically happen. Either status is bad and there's an alarm, or status is good and the second one never gets checked.

Reply:
Failure to read the MAC is not necessarily related to whether or not an interface is `UP` and/or `RUNNING`.

We're already tracking the case where with two interfaces connected, MotoROS2 always uses the first interface (ie: `ROS_USER_LAN1`) to get the MAC (#57) and to broadcast the debug logs (#58).

I've learned that failing early (and loudly) is better than trying to continue with whatever fallback does work.

Hence the question/comment.

> Also, if we do maintain this assertion, then the check for the second MAC will never logically happen. Either status is bad and there's an alarm, or status is good and the second one never gets checked.

yes, agreed.

0ec05f5ea8db598e4a855da72dda4a659fec165a was essentially just the old pattern but adapted to:

 - the changed `Ros_GetMacAddress(..)` implementation, and
 - no longer reuse alarm codes

Reply:
@ted-miller: would implementing the previous behaviour (ie: just try interfaces until one works) unblock this PR?

If so, I'll make that change and push it here.

We can update/correct later if desired.